packages/core/src/browser/shell/settings.tsx

```diff
@@ -0,0 +1,34 @@
+/********************************************************************************
+ * Copyright (C) 2020 Alibaba Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import { ReactWidget } from '../widgets';
+import { Emitter } from '../../common/event';
+
+/**
+ * The settings widget placed on the bottom of the left sidebar.
+ * Fire an onClick event when the widget onClick.
+ */
+export class Settings extends ReactWidget {
```

Comment:
We should give a better name `Widget` as a suffix for sure. But also it is not a settings widget, but rather a button to open it.

Reply:
`SettingsMenuWidget`